src/session_impl.cpp

```diff
@@ -5162,11 +5154,21 @@ namespace aux {
 		// proxy, and it's the same one as we're using for the tracker
 		// just tell the tracker the socks5 port we're listening on
 		if (m_socks_listen_socket && m_socks_listen_socket->is_open())
-			return m_socks_listen_socket->local_endpoint().port();
+		{
+			error_code ec;
+			tcp::endpoint endp = m_socks_listen_socket->local_endpoint(ec);
+			if (ec)
```

Comment:
sure, I will do that, let me revert the `BOOST_NO_EXCEPTIONS` changes.

I chose the wrong word, what I mean with "dangerous" is that I better get those exceptions and fail in java-land for better error reporting in the crash.